include/zephyr/drivers/espi.h

```diff
@@ -357,6 +365,10 @@ struct espi_request_packet {
  */
 struct espi_oob_packet {
 	uint8_t *buf;
+	/** For Tx packet, eSPI driver client shall specify length including total bytes
+	  * for eSPI OOB header and OOB payload.
```

Comment:
Consider additional clarification here, whether eSPI driver should exchange data from offset byte#3 in below OOB packet format or from byte#0. 

![image](https://github.com/zephyrproject-rtos/zephyr/assets/53013931/cc1b47ef-ff28-41e7-b5a2-7e41166196a7)

IMO, Byte#0,1,2 (eSPI cycle type, {tag, Len11:8}, Len 7:0) - aren't necessary.

Reply:
Added clarification indicating eSPI driver client and eSPI driver only exchange payload without OOB header